src/link/MachO.zig

```diff
@@ -150,6 +150,7 @@ no_implicit_dylibs: bool = false,
 /// Whether the linker should parse and always force load objects containing ObjC in archives.
 // TODO: in Zig we currently take -ObjC as always on
 force_load_objc: bool = true,
+rpaths: std.ArrayListUnmanaged([]const u8) = .{},
```

Comment:
Do we really need another array? An alternate strategy would be moving the logic from lines 424-245 to 786 and 3029 (`writeRpathLCs` looks safe to call twice).

Also, I'm confused why this is added to the MachO object. It seems like every call to `flushModule` will make it accumulate more and more rpaths. Was the intention to call `clearRetainingCapacity` in the beginning of `flushModule`?

Reply:
Mhm good idea about moving it into the functions in question. I shall do that.

Re calling flushModule multiple times - one step at a time :-) MachO is nowhere near incremental ready yet.